### PR TITLE
Colorize the consumer view

### LINKED ISSUE #48: Colorize the consumer view
The consumer view is pretty bleak:

consumer-view

Hopefully it should be possible to define and apply some textmate grammar, to give it more pizzaz

### PR DESCRIPTION
**Pull request by angelozerr**
Colorize the consumer view

Fixes #48

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
This PR colorize consumer view:

image

I need to improve my PR again, that's why I set it as draft.

This PR add a the new `kafka-consumer` language. I think the current `kafka `language should be renamed to `kafka-producer` to be consistent with `kafka-consumer`. @jlandersen if you like this idea, I could do that in this PR.

**Comment by jlandersen** [comment]
Nice addition!

Let's keep the language names - "kafka" is for user managed files and matches the .kafka filetype. I have some thought of being able to define consumers as well in the .kafka files at a later time.

An additional thought  - this might result in both kafka and kafka-consumer showing up in the language mode list, whereas kafka-consumer is never really meant to be something a user should see or be concerned with - would it be an idea to keep it in the same kafka language file?

**Comment by angelozerr** [comment]
> Nice addition!

Glad you like this PR!

> Let's keep the language names - "kafka" is for user managed files and matches the .kafka filetype. I have some thought of being able to define consumers as well in the .kafka files at a later time.

Ok let's keep kafka language.

> An additional thought - this might result in both kafka and kafka-consumer showing up in the language mode list, whereas kafka-consumer is never really meant to be something a user should see or be concerned with - would it be an idea to keep it in the same kafka language file?

IMHO I think it's a better idea to define 2 languages because for instance `Key: a-key` is not the same things for producer (ex : value of producer) and for consumer (it's the trace of consumer).

Having a kafka-consumer can be nice for the user who want to create a file from consumer trace and having syntax coloration.

What do you think about that?

**Comment by jlandersen** [comment]
> 
> 
> > Nice addition!
> 
> Glad you like this PR!
> 
> > Let's keep the language names - "kafka" is for user managed files and matches the .kafka filetype. I have some thought of being able to define consumers as well in the .kafka files at a later time.
> 
> Ok let's keep kafka language.
> 
> > An additional thought - this might result in both kafka and kafka-consumer showing up in the language mode list, whereas kafka-consumer is never really meant to be something a user should see or be concerned with - would it be an idea to keep it in the same kafka language file?
> 
> IMHO I think it's a better idea to define 2 languages because for instance `Key: a-key` is not the same things for producer (ex : value of producer) and for consumer (it's the trace of consumer).
> 
> Having a kafka-consumer can be nice for the user who want to create a file from consumer trace and having syntax coloration.
> 
> What do you think about that?

Ok i'll buy the part on setting it for a trace  :)

File: src/commands/consumers.ts [hunk 1]
```diff
@@ -58,6 +58,9 @@ export class StartConsumerCommandHandler {
                 preserveFocus: true,
                 viewColumn: vscode.ViewColumn.Beside,
             });
+        // Force the document language to kafka-consumer to benefit with syntax coloration
```
Comment (jlandersen) [inline src/commands/consumers.ts:61]: Just a small thing - I would understand it well from the API call, so comment might not be necessary

**Comment by jlandersen** [comment]
Just a single comment + please add this as an addition to the changelog for 0.10 then we'll get it merged :)

File: src/commands/consumers.ts (hunk 1 above)
Comment (angelozerr) [reply src/commands/consumers.ts]: Fixed, I removed the comment.

**Comment by angelozerr** [comment]
> Just a single comment + please add this as an addition to the changelog for 0.10 then we'll get it merged :)

Added

**Comment by jlandersen** [comment]
Cool, thanks!

**Comment by angelozerr** [comment]
Thanks @jlandersen !

**Pull request merged**